command/v3/v3_unshare_service_command_test.go

```diff
@@ -0,0 +1,233 @@
+
+	JustBeforeEach(func() {
+		executeErr = cmd.Execute(nil)
+	})
+
+	Context("when the API version is below the minimum", func() {
+		BeforeEach(func() {
+			fakeActor.CloudControllerAPIVersionReturns("0.0.0")
+		})
+
+		It("returns a MinimumAPIVersionNotMetError", func() {
+			Expect(executeErr).To(MatchError(translatableerror.MinimumAPIVersionNotMetError{
+				CurrentVersion: "0.0.0",
+				MinimumVersion: ccversion.MinVersionShareServiceV3,
+			}))
+		})
+	})
+
+	Context("when checking target fails", func() {
+		BeforeEach(func() {
+			fakeSharedActor.CheckTargetReturns(actionerror.NotLoggedInError{BinaryName: binaryName})
+		})
+
+		It("returns an error", func() {
+			Expect(executeErr).To(MatchError(actionerror.NotLoggedInError{BinaryName: binaryName}))
+
+			Expect(fakeSharedActor.CheckTargetCallCount()).To(Equal(1))
+			checkTargetedOrg, checkTargetedSpace := fakeSharedActor.CheckTargetArgsForCall(0)
+			Expect(checkTargetedOrg).To(BeTrue())
+			Expect(checkTargetedSpace).To(BeTrue())
+		})
+	})
+
+	Context("when the user is logged in, and a space and org are targeted", func() {
+		BeforeEach(func() {
+			fakeConfig.TargetedOrganizationReturns(configv3.Organization{
+				GUID: "some-org-guid",
+				Name: "some-org",
+			})
+			fakeConfig.TargetedSpaceReturns(configv3.Space{
+				GUID: "some-space-guid",
+				Name: "some-space",
+			})
+		})
+
+		Context("when getting the current user returns an error", func() {
+			var expectedErr error
+
+			BeforeEach(func() {
+				expectedErr = errors.New("get current user error")
+				fakeConfig.CurrentUserReturns(
+					configv3.User{},
+					expectedErr)
+			})
+
+			It("returns the error", func() {
+				Expect(executeErr).To(MatchError(expectedErr))
+			})
+		})
+
+		Context("when the current user is set correctly", func() {
+			BeforeEach(func() {
+				fakeConfig.CurrentUserReturns(
+					configv3.User{Name: "some-user"},
+					nil)
+			})
+
+			Context("when looking up the shared-to space guid returns an error", func() {
+				BeforeEach(func() {
+					fakeActorV2.GetSharedToSpaceGUIDReturns(
+						"",
+						v2action.Warnings{"get-shared-to-space-guid-warning"},
+						errors.New("an error"))
+				})
+
+				It("returns the error and displays all warnings", func() {
+					Expect(executeErr).To(MatchError("an error"))
+					Expect(testUI.Err).To(Say("get-shared-to-space-guid-warning"))
+				})
+			})
+
+			Context("when using the currently targeted org", func() {
+				BeforeEach(func() {
+					cmd.SpaceName = "some-space"
+					fakeActorV2.GetSharedToSpaceGUIDReturns(
+						"shared-to-space-guid",
+						v2action.Warnings{"get-shared-to-space-guid-warning"},
+						nil)
+				})
+
+				Context("when the unsharing is successful", func() {
+					BeforeEach(func() {
+						fakeActor.UnshareServiceInstanceFromSpaceReturns(
+							v3action.Warnings{"unshare-service-warning"},
+							nil)
+					})
+
+					It("unshares the service instance with the provided space and displays all warnings", func() {
```

Comment:
Please check GetSharedToSpaceGUID call count and args passed in on the first call

Reply:
Thanks, added a new test for this.